### ISSUE TITLE
[java] JUnitJupiterTestShouldBePackagePrivate: False negative when the only tests are in a @Nested class

### ISSUE THREAD

**Issue by UncleOwen** [pmd/pmd#6736]
**Affects PMD Version:** 7.25.0

**Rule:** [JUnitJupiterTestShouldBePackagePrivate](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#junitjupitertestshouldbepackageprivate)

**Code Sample demonstrating the issue:**

```java
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

public class FooTest extends SuperTest {
}

class SuperTest {
    @Nested
    class NestedClass {
        @Test
        void testFoo() {
        }
    }
}

```

**Expected outcome:** PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Diagnosis:** `TestFrameworksUtil.isJUnit5Class()` only checks for test methods, not for nested test classes. For that, we need a `JTypeMirror.streamNestedClasses()` (similar to `JTypeMirror.streamMethods()`)

**Running PMD through:** *Maven*
